backend/corpora/dataset_processing/download.py

```diff
@@ -171,6 +171,7 @@ def download(
             status = {
                 "upload_status": UploadStatus.FAILED,
                 "upload_message": str(progress_tracker.error),
+                "processing_status": ProcessingStatus.FAILURE,
```

Comment:
Updating the processing_status here rather than back in the processing.py code. It saves us another call to the DB where we were updating this value.